scripts/Global.gd

```diff
@@ -0,0 +1,82 @@
+extends Node
+
+onready var loading_scene = preload("res://scenes/LoadingScreen.tscn")
+var current_scene
+
+func _ready():
+	var root = get_tree().root
+	current_scene = root.get_child(root.get_child_count() - 1)
+
+# Use this function for loading in scenes which take a long time to load
+# next_scene: filepath to the scene you want
+func load_scene(next_scene):
+	# add loading scene to root
+	var loading_scene_instance = loading_scene.instance()
+	get_tree().get_root().call_deferred("add_child", loading_scene_instance)
+	
+	# start loading the next scene
+	var loader = ResourceLoader.load_interactive(next_scene)
+	# check for errors
+	if loader == null:
+		print("error occurred while getting the NEXT scene")
+		pass
+	
+	# remove current scene from root
+	current_scene.queue_free()
+	# wait for 0.5s for loading scene to appear
+	yield(get_tree().create_timer(1), "timeout")
+	
+	# load next scene using poll()
+	# poll() needs to be in a loop because it loads data in chunks
+	while true: 
+		var status = loader.poll()
+		# a chunk of data was loaded
+		if status == OK:
+			# update progress bar according to amount of data loaded
+			var progress_bar = loading_scene_instance.get_node("./CanvasLayer2/ProgressBar")
+			progress_bar.value = float(loader.get_stage()) / loader.get_stage_count() * 100
```

Comment:
Can you query and store `loader.get_stage_count()` outside the loop, and even store the inverse * 100 to avoid the frequent operations?

Reply:
(maybe this loop isn't run very often???)

Reply:
`status == OK:` is a situation that only happens when the loader is done loading a chunk of the scene, so it doesn't happen that often. I will still make those changes though.